src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,56 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.strategy.InterfaceImplementationFinderStrategy;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = InterfaceImplementationFinderStrategy.getImplementation(interfaceClazz);
+        Field[] fields = clazz.getDeclaredFields();
+        Object classImplementationInstance = null;
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                classImplementationInstance = createOrGetInstance(clazz);
+                setFieldToObject(field, classImplementationInstance, fieldInstance, clazz);
+            }
+        }
+        if (classImplementationInstance == null) {
+            classImplementationInstance = createOrGetInstance(clazz);
+        }
+        return classImplementationInstance;
+    }
+
+    private Object createOrGetInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Object instance = clazz.getConstructor().newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Cant create new instance of class " + clazz.getName(), e);
+        }
```

Comment:
Before attempting to create a new instance, let's check if the class has the `@Component` annotation. If it doesn't, we should throw an exception with an informative message. This is to ensure that we only create instances of classes that are marked as components.